### ISSUE TITLE
Suggestion: Map making

### ISSUE THREAD

**Issue by Drakaden** [zunath/CyberpunkZombieSurvival_JVM#155]
It has been something in the cauldron for a while but with no concrete method on how to do it yet, so i'll suggest a super simple procedure:

Add a variable to each area, named MapID, each map has its unique ID
Create 1 Map blueprint in the toolset
Player goes in a map, use map, map register MapID
Player use Map, Map is consumed, Map is registered under Key Items -> Map
Registered Key Map check area name of Map ID to token the name of the key item in the list
Whenever the player enter an area, check if he has mapID map, and if he does, reveal map.

Maps can be crafted at engineering station, as it is considerated as engineering to create such. A map could cost something as simple as 1 cloth.

**Comment by Drakaden** [comment]
I almost forgot, when adding MapID to map item, rename map to Map - Area Name, so each map is properly identified before using them up.

**Comment by taelon875** [comment]
I still disagree with this idea on the principle that it is the cartographer perk that lets maps be revealed.  If this was implemented on cartographers should be able to craft and it shouldn't be cheap.

**Comment by Drakaden** [comment]
Why shouldn't it be cheap? Map making isn't something that is material intensive, something to write on, a few pencils and maybe a ruler or two and yer good to go, though it should indeed be exclusive to cartographer to use the map, but anyone can make a blank cloth map.

**Comment by taelon875** [comment]
More of the fact that is completely negates one of the perks. So therefore it should be expensive.

**Comment by Drakaden** [comment]
Well you wanted to throw a "making sense" about AC, so i am making sense about map creating.

**Comment by taelon875** [comment]
I just say this shouldn't be a thing because it negates a perk. That is my final statement on this subject.

**Comment by Drakaden** [comment]
This perk you mention is still valid to the cartographer because only cartographers can make those maps, if anything the reveal map thing is a "you get to make those maps but you don't need to learn them"

**Comment by redcray0n** [comment]
I don't mind this idea, provided it's left for Cartographers only. Gives them the option to create extras and trade map information for goods or services. Shouldn't be a general craft, but something that is class-specific, if it was to go ahead.

**Comment by Drakaden** [comment]
Yes Red, that was the idea to leave the map making specific to cartographers, so that,s totally in line with what i suggested.

**Comment by zunath** [comment]
Professions have always been intended to have one temporary boost (which gets outpaced by gear, spells, or some other activity in game) and one permanent boost (which can never be replaced). 

In my mind, the temporary boost would be the revealing of maps automatically and the permanent boost would be the additional search attempt. So for that reason, I would see us adding a way to acquire maps for all players and not have it be restricted by the cartographer profession. These maps would be key items, so as not to take up inventory space.

**Comment by Drakaden** [comment]
Then we can simply make them craftable by everyone, it is the only variance in Zunath's thought.
